iidm/iidm-api/src/main/java/com/powsybl/iidm/network/InjectionBasedBusRef.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ */
+public class InjectionBasedBusRef implements BusRef {
```

Comment:
An afterthought : we could also have a simpler approach with only one implementation for all identifiables.
* in case of a configured bus ID, return the bus
* in case of an injection / bus bar, return the bus it belongs to
* otherwise, throw

It would have the benefit of reducing the number of implementations from 3 to 1 without too much loss (it will be slightly less "type safe" ...).